handler/handler_test.go

```diff
@@ -619,3 +599,30 @@ func TestWipeMetricStore(t *testing.T) {
 		}
 	}
 }
+
+// verifyMetricFamily jumps through a few hoops because the current protobuf
+// implementation is deliberately creating an unstable formatting for the text
+// representation. So this takes the text representation of the expected
+// MetricFamily and unmarshals it into a proto message object first. Then it
+// marshals both the expected and the got proto message into a binary protobuf,
+// which it then compares.
+func verifyMetricFamily(t *testing.T, expText string, got *dto.MetricFamily) {
```

Comment:
I wonder if we should add this helper somewhere in common.

Reply:
Yes, we should consider moving it up when it gets used there, too. However, I'm not quite sure yet if this is really the right way to do it… maybe some deep comparison without the final marshaling would be better?

Reply:
The recommended way I saw was to use https://pkg.go.dev/github.com/google/go-cmp/cmp to compare with https://pkg.go.dev/google.golang.org/protobuf/testing/protocmp.

Reply:
Oooof… two more dependencies instead of zero… I'd say we leave it as is in this PR and take some more time to consider our options for prometheus/common.

Reply:
Yea, I'm not a fan of what this change requires.